### PR TITLE
🎓 References Topic primitive types and lists in memory

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Show idealized views of how stuff can be stored in memory

### Why
Helps reinforce the idea of references and aliases if they have the intuition. 

### Additional Notes
Once again this is trying to cover something with nuance in a simple and straightforward way for new programmers. Lemmie' know if I am going to far one way or the other. Either way, this can be cleaned up again later if better ideas come along.

### CODE REVIEW

File: site/topics/references/references.rst [hunk 1]
```diff
@@ -18,90 +18,70 @@ References
...
-    * PROBLEM!
+    * This needs 256 bits
 
-* Wait, there's actually a simple solution. What if we block off chunks of RAM to be the array?
-* So if I have the array ``[a, b, c, d, e, f, g, h]``, we get this...
+* Fortunately we have a simple solution
+* Section off a large block of memory to store the contents of the list
+* For example, the following image shows how we could store the list ``[a, b, c, d, e, f, g, h]``
 
 .. image:: array_in_RAM.png
```
[the remark this answers is not in the document]
Comment (jameshughes89) [reply site/topics/references/references.rst:79]: I originally had this talking about "arrays" but opted to change it back to lists to not get too confusing.

[the remark this answers is not in the document]
Comment (twentylemon) [reply site/topics/references/references.rst:79]: I'm not sure what you're referring to, lists are arrays, doesn't really matter which you touch on here imo. My concern is the follow up question, what if the list contains different things? `collection_of_things = ["Hello", 10, True, 100.001]` is an example you've used in a previous topic.

[the remark this answers is not in the document]
Comment (twentylemon) [reply site/topics/references/references.rst:79]: a c b d

[the remark this answers is not in the document]
Comment (jameshughes89) [reply site/topics/references/references.rst:79]: I'll clean it up a little later today.

**Comment by jameshughes89** [comment]
I ended up just doing it now. 

A follow up PR will
- Finish the discussion
- Remind them that it's actually refs of refs
- Include a picture of refs of refs

**Review by twentylemon — APPROVED** [review]

**Pull request merged**